client/request.go

```diff
@@ -110,13 +110,20 @@ func (cli *Client) sendClientRequest(ctx context.Context, method, path string, q
 		if !cli.transport.Secure() && strings.Contains(err.Error(), "malformed HTTP response") {
 			return serverResp, fmt.Errorf("%v.\n* Are you trying to connect to a TLS-enabled daemon without TLS?", err)
 		}
+
 		if cli.transport.Secure() && strings.Contains(err.Error(), "remote error: bad certificate") {
 			return serverResp, fmt.Errorf("The server probably has client authentication (--tlsverify) enabled. Please check your TLS client certification settings: %v", err)
 		}
 
 		return serverResp, fmt.Errorf("An error occurred trying to connect: %v", err)
 	}
 
+	if resp != nil {
+		serverResp.statusCode = resp.StatusCode
+	}
+
+	defer resp.Body.Close()
```

Comment:
I wonder if this is a good way. When Close the resp.Body, serverResp.body will be influenced. 
And I think it is not a good way to define body in serverResponse as an  io.ReadCloser
```
type serverResponse struct {
	body       io.ReadCloser
	header     http.Header
	statusCode int
}
```
Usually we get the http response, get the body content using `ioutil.ReadAll` or other func, the Close the resp.Body.